atomic_reactor/plugins/pre_inject_yum_repos.py

```diff
@@ -0,0 +1,264 @@
+
+        final_user_line = self._final_user_line()
+        if final_user_line:
+            lines.insert(0, "USER root")
+            lines.append(final_user_line)
+
+        return lines
+
+    def add_koji_repo(self):
+        self.xmlrpc = get_koji_session(self.workflow.conf)
+        self.pathinfo = self.workflow.conf.koji_path_info
+        self.proxy = self.workflow.conf.yum_proxy
+
+        if not self.target:
+            self.log.info('no target provided, skipping plugin')
+            return
+
+        target_info = self.xmlrpc.getBuildTarget(self.target)
+        if target_info is None:
+            self.log.error("provided target '%s' doesn't exist", self.target)
+            raise RuntimeError("Provided target '%s' doesn't exist!" % self.target)
+        tag_info = self.xmlrpc.getTag(target_info['build_tag_name'])
+
+        if not tag_info or 'name' not in tag_info:
+            self.log.warning("No tag info was retrieved")
+            return
+
+        repo_info = self.xmlrpc.getRepo(tag_info['id'])
+
+        if not repo_info or 'id' not in repo_info:
+            self.log.warning("No repo info was retrieved")
+            return
+
+        # to use urljoin, we would have to append '/', so let's append everything
+        baseurl = self.pathinfo.repo(repo_info['id'], tag_info['name']) + "/$basearch"
+
+        self.log.info("baseurl = '%s'", baseurl)
+
+        repo = {
+            'name': 'atomic-reactor-koji-plugin-%s' % self.target,
+            'baseurl': baseurl,
+            'enabled': 1,
+            'gpgcheck': 0,
+        }
+
+        # yum doesn't accept a certificate path in sslcacert - it requires a db with added cert
+        # dnf ignores that option completely
+        # we have to fall back to sslverify=0 everytime we get https repo from brew so we'll surely
+        # be able to pull from it
+
+        if baseurl.startswith("https://"):
+            self.log.info("Ignoring certificates in the repo")
+            repo['sslverify'] = 0
+
+        if self.proxy:
+            self.log.info("Setting yum proxy to %s", self.proxy)
+            repo['proxy'] = self.proxy
+
+        yum_repo = YumRepo(os.path.join(YUM_REPOS_DIR, self.target))
+        path = yum_repo.dst_filename
+        self.log.info("yum repo of koji target: '%s'", path)
+        yum_repo.content = render_yum_repo(repo, escape_dollars=False)
+        self.yum_repos.append(yum_repo)
+
+    def __init__(self, workflow, target=None, repourls=None, inject_proxy=None):
+        """
+        constructor
+
+        :param workflow: DockerBuildWorkflow instance
+        :param target: string, koji target to use as a source
+        :param repourls: list of str, URLs to the repo files
+        :param inject_proxy: set proxy server for this repo
+        """
+        super().__init__(workflow)
+        self.target = target
+
+        self.xmlrpc = None
+        self.pathinfo = None
+        self.proxy = None
+        self.repourls = repourls or []
```

Comment:
repourls currently come from user_params? I think they should come from the resolve_composes plugin result instead (and they are different for each arch, so that may need more changes)

Reply:
yeah you should use yum_repourls from resolve_compose plugin

and add all noarch repos to all arches, and arch specific ones just to those arches

Reply:
updated to use `yum_repourls` from `resolve_composes` result